src/main/java/core/basesyntax/service/ProductParser.java

```diff
@@ -0,0 +1,15 @@
+package core.basesyntax.service;
+
+import core.basesyntax.model.Fruit;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public interface ProductParser {
+    Fruit parse(String productInfo);
+
+    default List<Fruit> parseAll(List<String> list) {
```

Comment:
didn't nee to create this method as default, move this logic directly to the implementation of parser